tempesta_fw/str.c

```diff
@@ -238,9 +244,11 @@ __cstricmpspn(const char *s1, const char *s2, int n, int stop)
 		c1 = tolower(*s1++);
 		c2 = tolower(*s2++);
 		if (c1 != c2)
-			return c1 < c2 ? -1 : 1;
-		if (!c1 || c1 == stop)
-			break;
+			return -1;
+		if (!c1)
+			return 0;
+		if (c1 == stop)
+			return 1;
```

Comment:
Why? Actually ```stop``` plays the same role as '\0', so why  the called need to know whether ```stop``` faced or not?